spyderlib/spyder.py

```diff
@@ -2660,6 +2675,218 @@ def show_tour(self, index):
+                        "your package manager to update Spyder or go to our "
+                        "<a href=\"{1}\">Releases</a> page to download this "
+                        "new version. <br><br>If you are not sure how to "
+                        "proceed to update Spyder please refer to our "
+                        " <a href=\"{2}\">Installation</a> instructions."
+                        "" .format(latest_release, url_r, url_i))
+                box.setText(msg)
+                box.set_check_visible(True)
+                box.exec_()
+                check_updates = box.is_checked()
+            elif feedback:
+                msg = _("Spyder is up to date.")
+                box.setText(msg)
+                box.set_check_visible(False)
+                box.exec_()
+                check_updates = box.is_checked()
+
+        # Update checkbox based on user interaction
+        CONF.set(section, option, check_updates)
+
+        # Enable check_updates_action after the thread has finished
+        self.check_updates_action.setDisabled(False)
+
+    def check_updates(self):
+        """
+        Check for spyder updates on github releases using a QThread.
+        """
+        # Disable check_updates_action while the thread is working
+        self.check_updates_action.setDisabled(True)
+
+        # feedback` = False is used on startup, so only positive feedback is
+        # given. `feedback` = True is used when using the menu action, and
+        # gives feeback if updates are, or are not found.
+        feedback = False
+
+        if DEV:
+            feedback = True
+        elif self.thread_updates is not None:
+            self.thread_updates.terminate()
+            feedback = True
+
+        self.thread_updates = QThread(self)
+        self.worker_updates = WorkerUpdates(self, feedback)
+        self.worker_updates.sig_ready.connect(self._check_updates_ready)
+        self.worker_updates.sig_ready.connect(self.thread_updates.quit)
+        self.worker_updates.moveToThread(self.thread_updates)
+        self.thread_updates.started.connect(self.worker_updates.check)
+        self.thread_updates.start()
+
+
+class MessageCheckBox(QMessageBox):
+    """
+
+    """
+    def __init__(self, *args, **kwargs):
+        super(MessageCheckBox, self).__init__(*args, **kwargs)
+
+        self._checkbox = QCheckBox()
+
+        # Access the Layout of the MessageBox to add the Checkbox
+        layout = self.layout()
+        layout.addWidget(self._checkbox, 1, 1, Qt.AlignRight)
+
+    # Methods to access the checkbox
+    def is_checked(self):
+        return self._checkbox.isChecked()
+
+    def set_checked(self, value):
+        return self._checkbox.setChecked(value)
+
+    def set_check_visible(self, value):
+        self._checkbox.setVisible(value)
+
+    def set_checkbox_text(self, text):
+        self._checkbox.setText(text)
+
+
+class WorkerUpdates(QObject):

```

Comment:
ok, but where?

I think `spyderlib\workers\updates.py` would make sense and keep it organized. This is not really a widget, and I think more workers will be needed when handling projects and version control to avoid blocking the UI.